lib/debug/session.rb

```diff
@@ -1038,53 +1043,44 @@ def self.open host: nil, port: ::DEBUGGER__::CONFIG[:port], sock_path: nil, sock
     else
       open_unix sock_path: sock_path, sock_dir: sock_dir
     end
+
+    set_starter_breakpoint
   end
 
   def self.open_tcp host: nil, port:, **kw
     set_config(kw)
     require_relative 'server'
-    initialize_session UI_TcpServer.new(host: host, port: port)
+    ui = UI_TcpServer.new(host: host, port: port)
+
+    if defined?(SESSION)
+      DEBUGGER__::SESSION.set_ui(ui)
```

Comment:
in scenarios like 

```ruby
require "debug"
a = 1
b = 2
DEBUGGER__.open
c = 3
d = 4
```

the debugger will first be initialized with `UI_Console`. and instead of initializing another session with a different ui, we can just assign it.